03-lists.md

```diff
@@ -86,6 +86,54 @@ TypeError: 'str' object does not support item assignment
 
 does not.
 
+> ## Do I share list memory or list values? {.callout}
+>
+> One thing we should know in python, while assigning a list A to another list B, the instruction B=A does assign A values to B, rather it shares A memory to B.
+> In other words, A and B have the same memory, which means any change in A concerns B and vice versa.Let see it through the following example:
+>
+>~~~ {.python}
+>A = [1, 4, 9, 25, 36]
+>B = A
+>print('A is :', A)
+>print('B is :', B)
+>~~~
+>
+>~~~ {.output}
+>A is: [1, 4, 9, 25, 36]
+>B is: [1, 4, 9, 25, 36]
+>~~~
+>
+>The change of the first value of A will change also the first value of B and the change of the fourth value of B will change the fourth value of A as well since they share the same memory location:
+>
+>~~~ {.python}
+>A[0]= 10
+>B[3]= 8
+>print('A is :', A)
+>print('B is :', B)
+>~~~
+>
+>~~~ {.output}
+>A is: [10, 4, 9, 8, 36]
+>B is: [10, 4, 9, 8, 36]
+>~~~
+>
+>All depends on the motivation of doing it. But if the motivation is to have a new list B with its own memory location, one of the ways is to do as follows: 
+>
+>~~~ {.python}
+>B=A[0:]
```

Comment:
Hi #dpshelio

Yeah,

B = A[:] and B = A[0:] give the same result,  B = A[0:5] as well.

I did not see where it was used before in "Programming with Python"
lesson that I read.
Can you show me the place, please?

Anyway, my contribution is in a sense that, compared to other
programming languages, when we assign
a variable to another, this creates a new location memory. But it is
not the case with Python.

Best regards.



On 5 July 2016 at 22:23, dpshelio <notifications@github.com> wrote:

> In 03-lists.md
> <https://github.com/swcarpentry/python-novice-inflammation/pull/280#discussion_r69630973>
> :
>
> > +>~~~ {.python}
> > +>A[0]= 10
> > +>B[3]= 8
> > +>print('A is :', A)
> > +>print('B is :', B)
> > +>~~~
> > +>
> > +>~~~ {.output}
> > +>A is: [10, 4, 9, 8, 36]
> > +>B is: [10, 4, 9, 8, 36]
> > +>~~~
> > +>
> > +>All depends on the motivation of doing it. But if the motivation is to have a new list B with its own memory location, one of the ways is to do as follows:
> > +>
> > +>~~~ {.python}
> > +>B=A[0:]
>
> Wouldn't be the following the same?
>
> B = A[:]
>
> Is it easier to understand it using [0:]? I think neither of these
> notations is shown in the lesson before... maybe we should do:
>
> B = A[0:5]
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/swcarpentry/python-novice-inflammation/pull/280/files/081b4884dc0f213511566ec7cbc137db8194357a#r69630973>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ASbhgjMJV9Taej5L6wGqs4xw9qui8fOEks5qSr0ogaJpZM4JDskO>
> .
>